src/cswinrt/strings/WinRT.cs

```diff
@@ -444,6 +446,28 @@ void _UnsubscribeFromNative()
             _token.Value = 0;
         }
     }
+
+    internal unsafe class EventSource__EventHandler<T> : EventSource<System.EventHandler<T>>
+    {
+        internal EventSource__EventHandler(IObjectReference obj,
+            delegate* unmanaged[Stdcall]<System.IntPtr, System.IntPtr, out WinRT.EventRegistrationToken, int> addHandler,
+            delegate* unmanaged[Stdcall]<System.IntPtr, WinRT.EventRegistrationToken, int> removeHandler) : base(obj, addHandler, removeHandler)
+        {
+        }
+
+        override protected System.Delegate EventInvoke
+        {
+            get
+            {
+                System.EventHandler<T> handler = (System.Object obj, T e) =>

```

Comment:
Should we be caching this or creating it each time?

Reply:
We can do that, but I'm not sure if that would give any significant perf gain

Reply:
My concern was more about whether each time we marshal EventInvoke, whether we are marshaling a new one or the same one.  Looking more at it, it looks like we only call EventInvoke the first time and then we don't so it might be fine.

Reply:
yes, EventInvoke is cached and then conditionally dispatches through to the end delegate accumulator.